contracts/Gambit.sol

```diff
@@ -21,23 +21,21 @@ contract Gambit is Token, Owned {
 
   // Get the total of token burnt
   /// @return Total amount of burned tokens
-  function totalBurnt() constant returns (uint totalBurnt) {
+  function totalBurnt() constant returns (uint256) {
     return _totalBurnt;
   }
 
   // Only the Owner of the contract can burn tokens.
   /// @param _value The amount of token to be burned
   /// @return Whether the burning was successful or not
-  function burn(uint _value) onlyOwner returns (bool success) {
-    if (balances[msg.sender] >= _value
-        && _value > 0) {
-      balances[msg.sender] -= _value;
-      _totalSupply         -= _value;
-      _totalBurnt          += _value;
-      Burn(msg.sender, _value);
-      return true;
-    } else {
-      return false;
-    }
+  function burn(uint256 _value) onlyOwner returns (bool) {
+    require(balances[msg.sender] >= _value);
```

Comment:
Previously you only did this action if `balances[msg.sender] >= _value`, but now you only do it if it's `< _value`. The same applies for the next line